src/Microsoft.OData.Core/UriParser/ODataQueryOptionParser.cs

```diff
@@ -474,19 +469,43 @@ private OrderByClause ParseOrderByImplementation(string orderBy, ODataUriParserC
             var orderByQueryTokens = expressionParser.ParseOrderBy(orderBy);
 
             // Bind it to metadata
+            BindingState state = CreateBindingState(configuration, odataPathInfo);
+
+            MetadataBinder binder = new MetadataBinder(state);
+            OrderByBinder orderByBinder = new OrderByBinder(binder.Bind);
+            OrderByClause orderByClause = orderByBinder.BindOrderBy(state, orderByQueryTokens);
+
+            return orderByClause;
+        }
+
+        private BindingState CreateBindingState(ODataUriParserConfiguration configuration, ODataPathInfo odataPathInfo)

```

Comment:
>CreateBindingState [](start = 29, length = 18)

What state from QueryOptionParser does CreateBindingState require that forces it to be an instance method, rather than a static method? applyClause and computeClause?

Reply:
Yes, applyCluase and computeClause are instance level